components/org.wso2.carbon.identity.remotefetch.common/src/main/java/org/wso2/carbon/identity/remotefetch/common/exceptions/RemoteFetchCoreException.java

```diff
@@ -33,4 +35,39 @@ public RemoteFetchCoreException(String message, Throwable cause) {
         super(message, cause);
     }
 
+    /**
+     * Constructs a new exception with the specified error code and cause.
+     *
+     * @param errorCode Error code
+     * @param message   Detailed message
+     */
+    public RemoteFetchCoreException(String errorCode, String message) {
```

Comment:
What is the expectation of the RemoteFetchCoreException and RemoteFetchServerException. When should these be used?

Reply:
RemoteFetchServerException is a child class of RemoteFetchCoreException. RemoteFetchServerException is used to point out a server error when processing user requests (OSGi service calls). 
RemoteFetchCoreException is used elsewhere to point out other exceptions.

Reply:
This particular method is to use to throw a new RemoteFetchServerException object(after handle server exception using Utils method.) with reference to its parent RemoteFetchCoreException.